rpc/connection.go

```diff
@@ -348,6 +333,27 @@ type ConnectionOpts struct {
 	DialerTimeout time.Duration
 }
 
+func NewTLSConnectionWithLogrus(
+	srvRemote Remote,
+	rootCerts []byte,
+	errorUnwrapper ErrorUnwrapper,
+	handler ConnectionHandler,
+	logFactory LogFactory,
+	logrusLogger LogrusLogger,
+	opts ConnectionOpts,
```

Comment:
Is there any chance the new logger argument could be added to the `ConnectionOpts` struct, instead of growing the rest of the API? The goal of that struct was that there could be a single place for non-mandatory args, and we could avoid the multiplicative growth of connection constructors.

Reply:
Hmm maybe not since it's replacing a mandatory argument. I just want to make sure we think about it.

Reply:
I guess we could move the unstructured one into `ConnectionOpts` as well, and just throw out an error if none is provided. Would you prefer that?

Reply:
I don't imagine there's a reasonable default we could assume if the logger wasn't provided? Some basic stdout/stderr thing? But anyway, I don't want to make you do more refactoring than you have to here.

Reply:
I guess as a portable package it could just use stderr for default. But it seems a bit confusing to have a default in addition to having two optional logger. I don't have strong opinion on this though.

Reply:
I think I'll go with the existing interface here. If it becomes too cluttered in the future we can change.